homeassistant/components/radarr/sensor.py

```diff
@@ -95,154 +76,137 @@
-        api_token=config[CONF_API_KEY],
-        ipaddress=config[CONF_HOST],
-        port=config[CONF_PORT],
-        ssl=config[CONF_SSL],
-        base_api_path=config[CONF_URLBASE],
-    )
-    radarr = RadarrClient(
-        host_configuration=host_configuration, session=async_get_clientsession(hass)
+    # deprecated in 2022.4
+    if "wanted" in config[CONF_MONITORED_CONDITIONS]:
+        LOGGER.warning(
+            "Wanted is not a valid condition option. Please remove it from your config"
+        )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+        )
     )
 
-    try:
-        await radarr.async_get_system_status()
-    except ArrException as ex:
-        raise PlatformNotReady() from ex
-    entities = [
-        RadarrSensor(config, radarr, description)
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Radarr sensors based on a config entry."""
+    async_add_entities(
+        RadarrSensor(hass.data[DOMAIN][entry.entry_id], description)
         for description in SENSOR_TYPES
-        if description.key in config[CONF_MONITORED_CONDITIONS]
-    ]
-    async_add_entities(entities, True)
+        if description.key != "wanted"
+    )
 
 
-class RadarrSensor(SensorEntity):
+class RadarrSensor(RadarrEntity, SensorEntity):
     """Implementation of the Radarr sensor."""
 
+    coordinator: RadarrDataUpdateCoordinator
+
     def __init__(
         self,
-        conf: ConfigType,
-        radarr: RadarrClient,
+        coordinator: RadarrDataUpdateCoordinator,
         description: SensorEntityDescription,
     ) -> None:
         """Create Radarr entity."""
+        super().__init__(coordinator)
         self.entity_description = description
-        self.radarr = radarr
-        self.included = cast(list, conf.get(CONF_INCLUDED))
-        self.days = int(cast(str, conf.get(CONF_DAYS)))
-        self.data: Any = None
         self._attr_name = f"Radarr {description.name}"
-        if description.key == "diskspace":
-            self._attr_native_unit_of_measurement = conf.get(CONF_UNIT)
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_{description.name}"
 
     @property
-    def extra_state_attributes(self) -> dict[str, StateType]:
+    def extra_state_attributes(self) -> dict[str, StateType | datetime]:
         """Return the state attributes of the sensor."""
-        attributes: dict[str, None | str | int | float] = {}
-        if self.entity_description.key == "upcoming":
-            return {to_key(movie): release_date(movie) for movie in self.data}
         if self.entity_description.key == "commands":
-            return {command.name: command.status for command in self.data}
+            return {cmd.name: cmd.status for cmd in self.coordinator.commands}
```

Comment:
What are commands ?

Reply:
The recently run commands in the application: download client status update, RSS sync, etc.

Reply:
those should not be in attributes.

Reply:
They should be separate sensors then and different PR in that case?

Reply:
Definitely not in the attributes. But why would we have that in HA at all ?

Reply:
I see your point so they will be removed. If someone in the community requests these, I can add them later as sensors.